open_catalyst_dataset/adsorptions.py

```diff
@@ -332,128 +361,202 @@ def tile_atoms(atoms):
+                                weights=weights, axis=0)
+
+    for idx, site in enumerate(slab_struct):
+        if site.frac_coords[2] > center_of_mass[2]:
+            try:
+                cn = voronoi_nn.get_cn(slab_struct, idx, use_weights=True)
+                cn = float('%.5f' % (round(cn, 5)))
+                # surface atoms are under-coordinated
+                if cn < min(bulk_cn_dict[site.species_string]):
+                    indices_list.append(idx)
+            except RuntimeError:
+                # or if pathological error is returned,
+                # indicating a surface site
+                indices_list.append(idx)
+    return indices_list
+
+
+
+def add_adsorbate_onto_surface(slab_atoms, surface_sites, adsorbate):
     '''
-    Randomly chooses an adsorbate from our pool of adsorbates.
+    There are a lot of small details that need to be considered when adding an
+    adsorbate onto a surface. This function will take care of those details for
+    you.
+
+    Args:
+        slab           An `ase.Atoms` object of the slab
+        surface_sites  A list of surface atoms' indices of the slab
+        adsorbate      An `ase.Atoms` object of the adsorbate
 
     Returns:
-        adsorbate   `ase.Atoms` object of the chosen adsorbate.
+        ads_slab       An `ase graphic Atoms` object containing the adsorbate and
+                       surface. Slab atoms will be tagged with a `0` and
+                       adsorbate atoms will be tagged with a `1`.
     '''
-    # TODO:  Kevin to add the real adsorbates here
-    CO = ase.Atoms('CO', positions=[[0., 0., 0.], [0., 0., 1.2]])
-    return CO
+    # convert slab atoms into graphic atoms object
+    slab_gratoms = catkit.Gratoms(slab_atoms)
+    slab_gratoms.set_surface_atoms(surface_sites)
+    slab_gratoms.pbc = np.array([True, True, False])
 
+    # set up the adsorbate into graphic atoms object
+    # with its connectivity matrix
+    adsorbate_gratoms = convert_adsorbate_atoms_to_gratoms(adsorbate['atoms_obj'])
+
+    # generate all possible adsorption configurations on that slab.
+    # The "bonds" argument automatically take care of mono vs.
+    # bidentate adsorption configuration.
+    builder = catkit.gen.adsorption.Builder(slab_gratoms)
+    adslabs = builder.add_adsorbate(adsorbate_gratoms, bonds=adsorbate['bond_idx'], index=-1)
+
+    # Filter out unreasonable structures.
+    # Then pick one from the reasonable configurations list as an output.
+    reasonable_adslabs = [adslab for adslab in adslabs if is_config_reasonable(adslab) is True]
+    ads_slab = random.choice(reasonable_adslabs)
+    return ads_slab
 
-def add_adsorbate_onto_surface(surface, adsorbate, site):
-    '''
-    There are a lot of small details that need to be considered when adding an
-    adsorbate onto a surface. This function will take care of those details for
-    you.
+
+def get_connectivity(adsorbate):
+    """
+    helper function that gets connectivity of an atoms obj.
```

Comment:
Don't need to say that it's a "helper function". It's apparent that it is a function. Just say what it does.